src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -40,15 +42,41 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<HashSet<Device>> {
         .collect()
 }
 
+#[derive(Debug)]
+pub struct BlkDevSegment {
+    pub blockdev: Rc<RefCell<BlockDev>>,
+    pub start: Sectors,
+    pub length: Sectors,
+}
+
+impl BlkDevSegment {
```

Comment:
```BlkDevSegment``` is strictly more informative than ```Segment```.

You can not implement ```From/Into``` for the two, obviously, but you should be able to implement a method on ```BlkDevSegment```, like ```into_segment()```. This ought to simplify ```convert_to_dm_segs()``` as well as a bunch of test code and maybe some other things as well.